apps/studio/lib/role-impersonation.ts

```diff
@@ -114,12 +112,63 @@ export function wrapWithRoleImpersonation(
   `
 }
 
-export function getRoleImpersonationJWT(
+export async function getRoleImpersonationJWT(
   projectRef: string,
   jwtSecret: string,
   role: PostgrestImpersonationRole
-) {
+): Promise<string> {
   const claims = getPostgrestClaims(projectRef, role)
 
-  return jwt.encode(claims, jwtSecret, 'HS256')
+  return createToken(claims, jwtSecret)
+}
+
+async function createToken(payload: object, key: string) {
+  var header = { typ: 'JWT', alg: 'HS256' }
+
+  var segments = []
+  segments.push(btoa(JSON.stringify(header)).replace(/=/g, ''))
+  segments.push(btoa(JSON.stringify(payload)).replace(/=/g, ''))
+
+  var footer = await sign(segments.join('.'), btoa(key).replace(/=/g, ''))
+
+  segments.push(footer.replace(/=/g, ''))
+
+  return segments.join('.')
+}
+
+async function sign(data: string, key: string) {
+  return window.crypto.subtle
+    .importKey(
+      'jwk',
+      {
+        //this is an example jwk key, "raw" would be an ArrayBuffer
+        kty: 'oct',
+        k: key,
+        alg: 'HS256',
+      },
+      {
+        name: 'HMAC',
+        hash: { name: 'SHA-256' },
+      },
+      false,
+      ['sign', 'verify']
+    )
+    .then((key) => {
+      var jsonString = JSON.stringify(data)
```

Comment:
likewise `const`s around here, and in the then block below

also quick question - any error handling (try catches) we need to do here?

Reply:
Using const now

At least added a console error log - it worked fine on my end, and canIUse support looks good for these functions